### PR TITLE
Allow map clones as valid map folders

### LINKED ISSUE #703: Map Making work flow - game engine pick up map clones.
It's very ackward to download a map, expand it, modify files, load it in the game, and then to repost the zip file somewhere.

I'm thinking we should update the game engine, so if you cloned a map repo directly into 'downloadMaps', that the game engine would pick up the map.

The change needed to the engine is so that it will look inside of a 'map' folder for game files.

### PR DESCRIPTION
**Pull request by DanVanAtta**
Resolves #703 

Updates to pick up map contents from a "map" folder if present. This allows us to work on 'git clones' directly from the downloadedMaps folder

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/905)

### CODE REVIEW

File: src/games/strategy/engine/framework/ui/NewGameChooserModel.java [hunk 1]
```diff
@@ -39,7 +39,14 @@
 
 
   public NewGameChooserModel() {
-    populate();
+    final Set<NewGameChooserEntry> parsedMapSet = parseMapFiles();
+
+    final List<NewGameChooserEntry> entries = Lists.newArrayList(parsedMapSet);
```
Comment (RoiEXLab) [inline src/games/strategy/engine/framework/ui/NewGameChooserModel.java]: According to guava this should be treatet as deprecated... Use new ArrayList<>(parsedMapset); instead...

File: src/games/strategy/triplea/ResourceLoader.java [hunk 2]
```diff
@@ -98,14 +99,16 @@ public boolean match(final File f) {
...
       }
     });
-    if (existing.size() > 1) {
-      System.out.println("INFO: Found too many files for: " + mapName + "  found: " + existing);
-    }
+
+
     // At least one must exist
     if (existing.isEmpty()) {
       return Collections.emptyList();
```
Comment (RoiEXLab) [inline src/games/strategy/triplea/ResourceLoader.java]: Same here... Use new ArrayList<>(); instead

File: src/games/strategy/triplea/ResourceLoader.java [hunk 3]
```diff
@@ -98,14 +99,16 @@ public boolean match(final File f) {
...
     }
     final File match = existing.iterator().next();
+    if (existing.size() > 1) {
```
Comment (RoiEXLab) [inline src/games/strategy/triplea/ResourceLoader.java]: @DanVanAtta you still didn't got the error.
This is occuring, even when there is only one map in one folder...
The Problem is that if `normalizedZipName` is equals `zipName` there are to *identical* entries in `existing`... e.g. `existing.size()` is greater than 1 although there is only a single Map file (written all lowercase with no spaces...)
#620 would have fixed that...

Comment (DanVanAtta) [inline src/games/strategy/triplea/ResourceLoader.java]: You're right, #620 would have fixed.  We also have a second problem, if the "map" folder exists, so will the parent. So in the git clone case, we will always get this warning.

File: src/games/strategy/engine/framework/ui/NewGameChooserModel.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/ui/NewGameChooserModel.java]: Just moving code around here. But okay, can fix it.

File: src/games/strategy/triplea/ResourceLoader.java (hunk 2 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ResourceLoader.java]: `Collections.emptyList();` is not guava. When the intent is for an empty collection, getting one from `Collections` is preferred to new'ing one up. It's similar to preferring `collection.isEmpty()` over `collection.size() > 0`

File: src/games/strategy/engine/framework/ui/NewGameChooserModel.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/ui/NewGameChooserModel.java]: Fixed it everywhere: 9c231bafcd

**Comment by DanVanAtta** [comment]
updated

File: src/games/strategy/triplea/ResourceLoader.java [hunk 4]
```diff
@@ -92,28 +90,16 @@ protected static String normalizeMapZipName(String zipName) {
...
-
-
-    // At least one must exist
-    if (existing.isEmpty()) {
-      return Collections.emptyList();
-    }
-    final File match = existing.iterator().next();
-    if (existing.size() > 1) {
-      ClientLogger.logQuietly("INFO: Found multiple files for: " + mapName + " using: "+ match + ", candidates found: " + existing);
+    Optional<File> match = candidates.stream().filter(file -> file.exists()).findFirst();
```
Comment (RoiEXLab) [inline src/games/strategy/triplea/ResourceLoader.java]: Java 8 Is great :heart_eyes:

File: test/games/strategy/engine/framework/map/download/MapDownloadListSortTest.java [hunk 5]
```diff
@@ -62,7 +62,7 @@ public void testInsertEmptyRowAboveHeaders() {
   @Test
   public void testSortingUnSortedListWithHeaders() {
     List<DownloadFileDescription> downloads =
-        new ArrayList<>(HEADER, MAP_B, MAP_A, HEADER, MAP_D, MAP_C, HEADER, HEADER);
+    Lists.newArrayList(HEADER, MAP_B, MAP_A, HEADER, MAP_D, MAP_C, HEADER, HEADER);
```
Comment (RoiEXLab) [inline test/games/strategy/engine/framework/map/download/MapDownloadListSortTest.java]: @DanVanAtta use `Arrays.asList()` instead
In all those cases

**Comment by RoiEXLab** [comment]
@DanVanAtta looks good now... See my comment about the ArrayList thing (marked as outdated)

**Pull request merged**